x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_lifecycle/index.tsx

```diff
@@ -0,0 +1,204 @@
+      actions.push({
+        name: i18n.translate('xpack.streams.streamDetailLifecycle.resetToDefault', {
+          defaultMessage: 'Reset to default',
+        }),
+        action: LifecycleEditAction.Inherit,
+      });
+    }
+
+    return actions;
+  }, [definition, isServerless]);
+
+  return {
+    lifecycleActions,
+    openEditModal,
+    setOpenEditModal,
+    updateInProgress,
+    setUpdateInProgress,
+  };
+}
+
+export function StreamDetailLifecycle({
+  definition,
+  refreshDefinition,
+}: {
+  definition?: IngestStreamGetResponse;
+  refreshDefinition: () => void;
+}) {
+  const {
+    core: { http, notifications },
+    dependencies: {
+      start: {
+        share,
+        streams: { streamsRepositoryClient },
+      },
+    },
+    isServerless,
+  } = useKibana();
+
+  const {
+    lifecycleActions,
+    openEditModal,
+    setOpenEditModal,
+    updateInProgress,
+    setUpdateInProgress,
+  } = useLifecycleState({ definition, isServerless });
+
+  const { signal } = useAbortController();
+
+  if (!definition) {
+    return null;
+  }
+
+  const ilmLocator = share.url.locators.get<IlmLocatorParams>(ILM_LOCATOR_ID);
+
+  const getIlmPolicies = async () => {
+    const response = await http.get<PolicyFromES[]>('/api/index_lifecycle_management/policies', {
+      signal,
+    });
+    return response;
+  };
+
+  const updateLifecycle = async (lifecycle: IngestStreamLifecycle) => {
+    try {
+      setUpdateInProgress(true);
+
+      const request = {
+        ingest: {
+          ...definition.stream.ingest,
+          lifecycle,
+        },
+      } as IngestUpsertRequest;
+
+      await streamsRepositoryClient.fetch('PUT /api/streams/{id}/_ingest', {
+        params: {
+          path: { id: definition.stream.name },
+          body: request,
+        },
+        signal,
+      });
+
+      refreshDefinition();
+      setOpenEditModal(LifecycleEditAction.None);
+
+      notifications.toasts.addSuccess({
+        title: i18n.translate('xpack.streams.streamDetailLifecycle.updated', {
+          defaultMessage: 'Stream lifecycle updated',
+        }),
+      });
+    } catch (error) {
+      notifications.toasts.addError(error, {
+        title: i18n.translate('xpack.streams.streamDetailLifecycle.failed', {
+          defaultMessage: 'Failed to update lifecycle',
+        }),
+        toastMessage: 'body' in error ? error.body.message : error.message,
+      });
+    } finally {
+      setUpdateInProgress(false);
+    }
+  };
+
+  return (
+    <>
+      <EditLifecycleModal
+        action={openEditModal}
+        definition={definition}
+        closeModal={() => setOpenEditModal(LifecycleEditAction.None)}
+        updateLifecycle={updateLifecycle}
+        getIlmPolicies={getIlmPolicies}
+        updateInProgress={updateInProgress}
+        ilmLocator={ilmLocator}
+      />
+
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup direction="column" gutterSize="s">
```

Comment:
do we need this FlexGroup/FlexItem ? Seems like it's not doing anything, but I might be wrong, didn't test